akka-remote/src/main/scala/akka/remote/artery/AeronSink.scala

```diff
@@ -2,60 +2,101 @@ package akka.remote.artery
 
 import java.nio.ByteBuffer
 import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicInteger
 
 import scala.annotation.tailrec
 import scala.concurrent.duration._
 
 import akka.stream.Attributes
 import akka.stream.Inlet
 import akka.stream.SinkShape
+import akka.stream.stage.AsyncCallback
 import akka.stream.stage.GraphStage
 import akka.stream.stage.GraphStageLogic
 import akka.stream.stage.InHandler
-import akka.stream.stage.TimerGraphStageLogic
 import io.aeron.Aeron
+import io.aeron.Publication
 import org.agrona.concurrent.BackoffIdleStrategy
 import org.agrona.concurrent.UnsafeBuffer
 
 object AeronSink {
   type Bytes = Array[Byte]
-  private case object Backoff
+
+  private def offerTask(pub: Publication, buffer: UnsafeBuffer, msgSize: AtomicInteger, onOfferSuccess: AsyncCallback[Unit]): () ⇒ Boolean = {
```

Comment:
It's not the activation signal, but it must be updated when handing over the task to the runner, since it represents the bytes to take from the buffer. I also think that is ugly, but I couldn't think of other way that didn't involve allocations.